packages/lib/server/checkBookingLimits.ts

```diff
@@ -0,0 +1,89 @@
+import dayjs from "@calcom/dayjs";
+import prisma from "@calcom/prisma";
+import { BookingLimit } from "@calcom/types/Calendar";
+
+import { HttpError } from "../http-error";
+
+export async function checkBookingLimits(
+  bookingLimits: BookingLimit,
+  eventStartDate: Date,
+  eventId: number,
+  returnBusyTimes?: boolean
+) {
+  const limitCalculations = Object.entries(bookingLimits).map(
+    async ([key, limitingNumber]) =>
+      await checkLimit({ key, limitingNumber, eventStartDate, eventId, returnBusyTimes })
+  );
+  await Promise.all(limitCalculations)
+    .then((res) => {
+      if (returnBusyTimes) {
```

Comment:
This is used if we want to return the value - not always the case -> This is used when we calculate availability